src/core/server/legacy/plugins/get_nav_links.ts

```diff
@@ -38,6 +38,7 @@ function legacyAppToNavLink(spec: LegacyAppSpec): LegacyNavLink {
     euiIconType: spec.euiIconType,
     url: spec.url || `/app/${spec.id}`,
     linkToLastSubUrl: spec.linkToLastSubUrl ?? true,
+    href: undefined,
```

Comment:
Not sure what is going on here (or below), hoping someone can shine some light on this? Is this ok?

`ChromeNavLink` complained of a type failure without this but I don't really understand when this object is used or how much of it is important... Seems like a lot of it is rebuilt in _to_nav_link.ts_ or in _nav_links_service.ts_ (not touched in this PR)...

Reply:
This is caused by the `LegacyNavLink` definition:

https://github.com/elastic/kibana/blob/abbb51ba11feb2c24e61e07969773ea73e56eadd/src/core/server/legacy/types.ts#L154-L156

You should probably just add `href` to the list of omitted property from this def.

> I don't really understand when this object is used or how much of it is important

This is the (first part of the) legacy navlink to KP navlink bridge.